### ISSUE TITLE
Inifinite loop in the DateTime resolvers.

### ISSUE THREAD

**Issue by glennawatson** [reactiveui/Akavache#463]
My pr introduced a bug showing an infinite loop of calls exhausting the stack:   
edit: it's not my pr, it's your code lol :)
edit2: it's not my pr, it's not your code. It's when an existingContractResolver is non null and derives from DefaultContractResolver. Then it loops ...

_Originally posted by @softlion in https://github.com/reactiveui/Akavache/pull/459#issuecomment-464832373_

```
Newtonsoft_Json_Utilities_ThreadSafeStore_2_TKey_REF_TValue_REF_Get_TKey_REF + 29119496 (ThreadSafeStore.cs:64)
Newtonsoft_Json_Serialization_DefaultContractResolver_ResolveContract_System_Type + 29123632 (DefaultContractResolver.cs:198)
Akavache_JsonDateTimeContractResolver_CreateContract_System_Type + 38540100 (/D:\a\1\s\src\Akavache.Core\JsonDateTimeContractResolver.cs:23)
System_Collections_Concurrent_ConcurrentDictionary_2_TKey_REF_TValue_REF_GetOrAdd_TKey_REF_System_Func_2_TKey_REF_TValue_REF + 180
Newtonsoft_Json_Utilities_ThreadSafeStore_2_TKey_REF_TValue_REF_Get_TKey_REF + 29119496 (ThreadSafeStore.cs:64)
Newtonsoft_Json_Serialization_DefaultContractResolver_ResolveContract_System_Type + 29123632 (DefaultContractResolver.cs:198)
Akavache_JsonDateTimeContractResolver_CreateContract_System_Type + 38540100 (/D:\a\1\s\src\Akavache.Core\JsonDateTimeContractResolver.cs:23)
System_Collections_Concurrent_ConcurrentDictionary_2_TKey_REF_TValue_REF_GetOrAdd_TKey_REF_System_Func_2_TKey_REF_TValue_REF + 180
Newtonsoft_Json_Utilities_ThreadSafeStore_2_TKey_REF_TValue_REF_Get_TKey_REF + 29119496 (ThreadSafeStore.cs:64)
Newtonsoft_Json_Serialization_DefaultContractResolver_ResolveContract_System_Type + 29123632 (DefaultContractResolver.cs:198)
Akavache_JsonDateTimeContractResolver_CreateContract_System_Type + 38540100 (/D:\a\1\s\src\Akavache.Core\JsonDateTimeContractResolver.cs:23)
System_Collections_Concurrent_ConcurrentDictionary_2_TKey_REF_TValue_REF_GetOrAdd_TKey_REF_System_Func_2_TKey_REF_TValue_REF + 180
Newtonsoft_Json_Utilities_ThreadSafeStore_2_TKey_REF_TValue_REF_Get_TKey_REF + 29119496 (ThreadSafeStore.cs:64)
Newtonsoft_Json_Serialization_DefaultContractResolver_ResolveContract_System_Type + 29123632 (DefaultContractResolver.cs:198)
Akavache_JsonDateTimeContractResolver_CreateContract_System_Type + 38540100 (/D:\a\1\s\src\Akavache.Core\JsonDateTimeContractResolver.cs:23)
System_Collections_Concurrent_ConcurrentDictionary_2_TKey_REF_TValue_REF_GetOrAdd_TKey_REF_System_Func_2_TKey_REF_TValue_REF + 180
Newtonsoft_Json_Utilities_ThreadSafeStore_2_TKey_REF_TValue_REF_Get_TKey_REF + 29119496 (ThreadSafeStore.cs:64)```

**Comment by glennawatson** [comment]
This is related to a issue that @softlion brought up in #459

**Comment by glennawatson** [comment]
I think the easy solution is to check if the resolver is derived from the DefaultContractResolver to prevent the re-entry. I won't realistically get it to later this week so if you get a PR in before then I will merge as soon as I see it.

**Comment by softlion** [comment]
should be resolved in #465 

I can't reproduce the loop in unit tests. but i can in the real app (randomly -often enough- though).

**Comment by softlion** [comment]
Plz mege and post nuget as soon as you can, its very urgent ;)
